### PR TITLE
Adding registration enumeration to the EnrollmentGroup provisioning sample

### LINKED ISSUE #424: CreateEnrollmentGroupQuery with query.NextAsync() returns group info without the list of devices contained in the group
- **OS, version, SKU and CPU architecture used:** <VERSION>  Windows 10
- **Application's .NET Target Framework :** <NETVERSION>  .Net 4.6
- **Device:** <DEVICETYPE> 
- **SDK version used:** <VERSION>  Latest version from the Nuget

# Description of the issue:

I have exposed custom apis for DPS using DPS SDK. There is an api which returns the groups from the DPS. I observed that the when following code snippet is run I am getting the group info but not the devices within the group which is kinda counter intuitive. Can you please make changes or put a flag which can be set so that devices from the group are returned. Also please suggest any other ways via which I can retrieve the devices in a given group.

Please be advised that we need the the group info and the list of devices within the group to be exposed via api for our project which is already running on very aggressive deadlines.

# Code sample exhibiting the issue:

           Query g = _helper.CreateEnrollmentGroupsQuery(dpsConnectionString, querySpecification, pageSize);
           QueryResult queryResult = await g.NextAsync().ConfigureAwait(false);
           
           public Query CreateEnrollmentGroupsQuery(string dpsConnectionString, QuerySpecification querySpecification, int datapageSize = 0)
        {

            var provisioningServiceClient = ProvisioningServiceClient.CreateFromConnectionString(dpsConnectionString);
            if (datapageSize != 0)
            {
                Query q = provisioningServiceClient.CreateEnrollmentGroupQuery(querySpecification, datapageSize);

                return q;
            }
            else
            {
                Query q = provisioningServiceClient.CreateEnrollmentGroupQuery(querySpecification);
                return q;
            }

        }

# Console log of the issue:

### PR DESCRIPTION
**Pull request by CIPop**
# Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [X] This pull-request is submitted against the `master` branch.

# Description of the changes
Adding registration enumeration sample code for Enrollment Groups.

# Reference/Link to the issue solved with this PR (if any)
Fixes #424

### CODE REVIEW

**Comment by CIPop** [comment]
@abhipsaMisra @zolvarga PTAL - I'm still having issues with GitHub not being able to populate the reviewers list...

File: provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs [hunk 1]
```diff
@@ -97,6 +101,24 @@ public static void Main(string[] args)
...
         }
 
+        /// <summary>
+        /// Enumerates all registrations within an enrollment group.
+        /// </summary>
+        /// <param name="provisioningServiceClient">The ProvisioningServiceClient object.</param>
+        /// <param name="querySpecification">The query specification.</param>
+        /// <param name="group">The EnrollmentGroup object.</param>
+        /// <returns></returns>
+        private static async Task EnumerateDevicesInGroup(ProvisioningServiceClient provisioningServiceClient, QuerySpecification querySpecification, EnrollmentGroup group)
```
Comment (alextolp) [inline provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs]: nit: EnumerateRegistrationsInGroup

File: provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs [hunk 2]
```diff
@@ -97,6 +101,24 @@ public static void Main(string[] args)
...
+        {
+            Console.WriteLine($"\nCreating a query for registrations within group '{group.EnrollmentGroupId}'...");
+            using (Query registrationQuery = provisioningServiceClient.CreateEnrollmentGroupRegistrationStateQuery(querySpecification, group.EnrollmentGroupId))
```
Comment (nberdy) [inline provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs]: Would be nice to have an example query specification so it's clear what the user needs to construct.

Comment (CIPop) [reply provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs]: @nberdy - we're passing the same one. I thought the service doesn't support anything else here.
/cc @alextolp

Comment (nberdy) [reply provisioning/service/samples/ProvisioningServiceEnrollmentGroup/Program.cs]: Ignore me, I missed the definition because it was collapsed in the PR

**Pull request merged**